backend/packages/Upgrade/src/api/services/ExperimentUserService.ts

```diff
@@ -295,6 +295,20 @@ export class ExperimentUserService {
     }
   }
 
+  private async safelyTrySaveUserDocuments(documents: any) {
```

Comment:
true, got lost in what I was trying to do. the intent was to just log the error and _avoid_ throwing, in order to log the error without stopping execution because the error that sometimes comes up here is that it is trying to insert at a key that already exists, which was the solution to issues in #710, some rare edge-case with `.save`.

I thought to do this would work, but the function would need a return value so that the execution can continue post-processing the documents for user response. Maybe it can just return the documents that were passed in? 

```
try {
      return this.userRepository.save(documents);
    } catch (error) {
      const documentSaveError = new Error(
        JSON.stringify({
          type: SERVER_ERROR.QUERY_FAILED,
          message: `Error while saving user documents: ${JSON.stringify(documents)}`,
          details: error,
        })
      );

      logger.error(documentSaveError);
      // what to return?
    }
 ```
 
I also instead looked at what would be needed to do what you suggested in #710 for upserting with an "onConflict" but it looks like "onConflict" is deprecated. I'm not an expert on TypeORM but perhaps that is a better solution than what I'm trying to do, which is to just log the error and keep going, if you can provide an example for a doing upsert while avoiding `.save`.

Reply:
A common function won't work over here as some return an object and some array of objects.

Reply:
@danoswaltCL are you expecting this type of code?
Insert - 

```
const result = await this.createQueryBuilder('user')
      .insert()
      .values(userArray)
      .returning('*')
      .execute()
      .catch((errorMsg: any) => {
        const errorMsgString = repositoryError('UserRepository', insertUsers', { userDocs }, errorMsg);
        throw errorMsgString;
      });

    return result.raw;
   
``` 
   
Update - 
```
const result = await this.createQueryBuilder('user')
      ..update()
      .set(userDoc)
      .where("id = :id", { id: userDoc.id })
      .returning('*')
      .execute()
      .catch((errorMsg: any) => {
        const errorMsgString = repositoryError('UserRepository', 'updateUser', { userDoc }, errorMsg);
        throw errorMsgString;
      });

    return result.raw;
```

If want to combine both, then we need to use onConflict

```
const result = await this.createQueryBuilder('user')
      .insert()
      .values(userDoc)
      .onConflict(`("id") DO UPDATE SET name = userDoc.name, ..., ...)
      .returning('*')
      .execute()
      .catch((errorMsg: any) => {
        const errorMsgString = repositoryError('UserRepository', 'updateUser', { userDoc }, errorMsg);
        throw errorMsgString;
      });

return result.raw;
```